js/walutomat.js

```diff
@@ -0,0 +1,275 @@
+        let markets = [];
+        let currenciesLength = currencies.length;
+        for (let base = 0; base < currenciesLength - 1; base++) {
+            let newIndex = base + 1;
+            for (let quote = newIndex; quote < currencies.length; quote++) {
+                markets.push ({
+                    'id': currencies[base] + '_' + currencies[quote],
+                    'symbol': currencies[base] + '/' + currencies[quote],
+                    'base': currencies[base],
+                    'quote': currencies[quote],
+                    'active': true,
+                    'precision': {
+                        'amount': 2,
+                        'price': 4,
+                    },
+                    'info': currencies[base] + ' - ' + currencies[quote] + ' exchange',
+                });
+            }
+        }
+        return markets;
+    }
+
+    async fetchBalance (params = {}) {
+        let balances = await this.privateGetAccountBalances ();
+        let result = {};
+        for (let i = 0; i < balances.length; i++) {
+            let balance = balances[i];
+            result[this.safeString (balance, 'currency')] = {
+                'free': this.safeFloat (balance, 'balanceAvailable'),
+                'used': this.safeFloat (balance, 'balanceReserved'),
+                'total': this.safeFloat (balance, 'balanceAll'),
+            };
+        }
+        return this.parseBalance (result);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let symbolObject = { 'symbol': this.marketId (symbol) };
+        let response = await this.publicGetPublicMarketOrderbookSymbol (this.extend (symbolObject, params));
+        return this.parseOrderBook (response, undefined, 'bids', 'asks', 'price', 'baseVolume');
+    }
+
+    async fetchMyTrades (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let request = {
+            'operationType': 'MARKET_FX',
+            'sortOrder': 'DESC',
+        };
+        if (symbol !== undefined) {
+            request['currencies'] = this.marketId (symbol);
+        }
+        if (since !== undefined) {
+            // TODO: Not sure if it's correct (idk if it's timestamp or DateTime string
+            request['continueFrom'] = since;
+        }
+        if (limit !== undefined) {
+            request['volume'] = limit;
+        }
+        let response = await this.privateGetAccountHistory (this.extend (request, params));
+        let result = [];
+        for (let i = 0; i < response.length; i++) {
+            result.push ({
+                'id': this.safeInteger (response[i], 'id'),
+            });
+        }
+        return result;
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        let pair = this.marketId (symbol);
```

Comment:
I suggest that we change and polish this particular method a bit ) Just to make it a clean and unified code style across exchanges...

```JavaScript
let pair = this.marketId (symbol);
let currencies = pair.split('_');
let body = {
    ...,
    'volumeCurrency': currencies[0],
    'otherCurrency': currencies[1],
};
```

↓

```JavaScript
const market = this.market (symbol);
const body = {
    ...,
    'volumeCurrency': market['baseId'],
    'otherCurrency': market['quoteId'],
};
```

I know this is very pedantic, and let me know if you don't want to do that, I'll try and do this myself asap. This probably applies not only to this method, but a few other spots throughout the file as well. If we keep with somewhat "unified" codestyle – we will be able to generalize common patterns across exchanges into the base class methods – we actually did that already, so, it really helps.

Reply:
Sure I will update everything, no worries, code quality is obviously very important! :)